test/langtools/tools/javac/api/taskListeners/TestTypeElement.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8283661
+ * @summary Unexpected TypeElement in ANALYZE TaskEvent
+ * @modules jdk.compiler
+ * @run main TestTypeElement
+ */
+
+import java.io.IOException;
+import java.io.PrintStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.List;
+
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.util.Elements;
+import javax.tools.JavaCompiler;
+import javax.tools.JavaFileObject;
+import javax.tools.SimpleJavaFileObject;
+import javax.tools.ToolProvider;
+
+import com.sun.source.util.JavacTask;
+import com.sun.source.util.TaskEvent;
+import com.sun.source.util.TaskListener;
+
+public class TestTypeElement {
+    public static void main(String... args) throws Exception {
+        new TestTypeElement().run();
+    }
+
+    private PrintStream log;
+    private Elements elements;
+    private int errors;
+
+    void run() throws Exception {
+        log = System.err;
+
+        List<JavaFileObject> files = List.of(
+                createFileObject("module-info.java", "module m { }"),
+                createFileObject("p/package-info.java", "/** Comment. */ package p;"),
+                createFileObject("p/C.java", "package p; public class C { }")
+        );
+
+        JavaCompiler c = ToolProvider.getSystemJavaCompiler();
+        JavacTask t = (JavacTask) c.getTask(null, null, null,  List.of("-d", "classes"), null, files);
+        t.addTaskListener(new TaskListener() {
+            @Override
+            public void started(TaskEvent e) {
+                log.println("started: " + e);
```

Comment:
you don't seem to be processing this output later on, I guess it could be removed, unless you want to keep it for debug reasons

Reply:
yeah, it's helpful to have the printed information in case things go wrong.

Reply:
Yeah, it is a concise way of displaying the problems, if any.